frontend/dockerfile/dockerfile2llb/directives.go

```diff
@@ -38,8 +39,15 @@ func ParseDirectives(r io.Reader) map[string]Directive {
 	var l int
 	for s.Scan() {
 		l++
+		if reShebang.MatchString(s.Text()) {
```

Comment:
I think instead of changing the `ParseDirectives` we should add the custom cases in `DetectSyntax()`. If `directives[keySyntax]` is not found then we check the custom cases. Otherwise, Dockerfile directive rules do not change.

For example this allows rules like shebang is required to be on the first line only and shebang is still not allowed on regular Dockerfiles.